### PR TITLE
feat: Add QR code scanner for lightning invoices

### PR DESCRIPTION
**Pull request by klochowicz**
QR code scanner opens up in a separate widget. It fills in the lightning invoice
string if a valid QR code is found. Validation of the invoice is handled in the
send widget, as well as showing the details of the invoice.

Note: it should be fairly trivial to do the same for bitcoin QR codes, I will
proceed with it after this PR gets in.

here's streaming of the app from my iPhone (can't really do this from the simulator):

https://user-images.githubusercontent.com/8319440/204969909-5779eedd-5c94-4bb7-a0c3-a99085ffd367.mov

### CODE REVIEW

**Comment by klochowicz** [comment]
right now the widget overflows, but can parse an invoice and fill it :)

**Comment by holzeis** [comment]
hey @klochowicz, I tested this on my android device. Looks nice already. 🙂 

fyi: I had to cherry pick a47eb3df1a00259e7664754bf2a29448becd3e28 and update the `minSdkVersion` to 21 in the build gradle to build on android.

**Review by bonomat — COMMENTED** [review]
A video or screenshots would be helpful :) I'm not good enough in dart yet to provide other feedback... but I guess, you are not yet asking for feedback? :D

**Comment by klochowicz** [comment]
> hey @klochowicz, I tested this on my android device. Looks nice already. 🙂 This could also be useful on the corresponding bitcoin view.
> 
> fyi: I had to cherry pick [a47eb3d](https://github.com/itchysats/10101/commit/a47eb3df1a00259e7664754bf2a29448becd3e28) and update the `minSdkVersion` to 21 in the build gradle to build on android.

thanks for testing this on android! I will cherry pick that commit too into my changeset. 

I'm just reworking it into it's own widget that can be used for scanning in both contexts. (activated on pressing `scan`, otherwise we're taking up a lot of screen real estate)

**Comment by klochowicz** [comment]
> A video or screenshots would be helpful :) I'm not good enough in dart yet to provide other feedback... but I guess, you are not yet asking for feedback? :D

the frontend is not ready yet, the most I hoped for was Android test, which I got :-) will attempt to record iOS operation

File: lib/wallet/send.dart [hunk 1]
```diff
@@ -122,6 +117,10 @@ class _SendState extends State<Send> {
                         Row(
                           mainAxisAlignment: MainAxisAlignment.end,
                           children: [
+                            TextButton.icon(
+                                onPressed: () => {GoRouter.of(context).go(QrScan.route)},
+                                icon: const Icon(Icons.qr_code_scanner),
+                                label: const Text("QR Scan")),
                             SubmitButton(
```
Comment (holzeis) [inline lib/wallet/send.dart:124]: nit: A little bit of space would be nice here. 

e.g.
const SizedBox(width: 5)

**Review by holzeis — APPROVED** [review]
Very nice! 👍 I guess the video was taken on a later version as the code, as I wouldn't print the encoded invoice in the snackbar 😉 .

File: lib/wallet/send.dart (hunk 1 above)
Comment (klochowicz) [reply lib/wallet/send.dart:125]: done :)

**Pull request merged**